### PR TITLE
[MM-37933] Channel preference to auto-follow all threads in the channel

### PR DESCRIPTION
**Pull request by vish9812**
#### Summary

Create the thread membership whenever there is a reply on a post.

#### Ticket Link

JIRA - https://mattermost.atlassian.net/browse/MM-37933

#### Release Note

```release-note
Added a new option to auto-follow all threads in the Channel's Notification Preference.
```

### CODE REVIEW

**Review by koox00 — DISMISSED** [review]

File: app/notification_test.go [hunk 1]
```diff
@@ -2821,3 +2822,74 @@ func TestReplyPostNotificationsWithCRT(t *testing.T) {
...
+	th.AddUserToChannel(u3, c1)
+
+	// Set auto-follow for user 2
+	member, appErr := th.App.UpdateChannelMemberNotifyProps(th.Context, map[string]string{model.ChannelAutoFollowThreads: "true"}, c1.Id, u2.Id)
+	require.Nil(t, appErr)
+	require.Equal(t, "true", member.NotifyProps[model.ChannelAutoFollowThreads])
+
+	// Enable CRT
+	os.Setenv("MM_FEATUREFLAGS_COLLAPSEDTHREADS", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_COLLAPSEDTHREADS")
```
Comment (ashishbhate) [inline app/notification_test.go:2845]: I don't think this is necessary any more.

**Review by ashishbhate — DISMISSED** [review]

**Comment by ashishbhate** [comment]
>Added a new option to auto-follow all threads in the Channel's Notification Preference.

AFAIK, the frontend for this feature hasn't been built yet. 
Is the PR not going to be merged until the frontend is done? If we're going to merge this PR early, then we need to change (or remove) the release note.

**Comment by esethna** [comment]
@Willyfrog ^

**Comment by Willyfrog** [comment]
Good catch @ashishbhate 
Yes we should remove it until it is accesible.
@AshishDhama remember to add it to your PR when you add the UI for this

**Comment by jgilliam17** [comment]
Thanks you @vish9812 
I checked [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/253021-68ab16c-onprem-ent-server-pr-21430-mm-ee-test:68ab16c/mochawesome.html) - looks good, no PR related failures
Let me know when I can test the actual menu options on the test server? thanks 🙂

File: app/notification_test.go (hunk 1 above)
Comment (vish9812) [reply app/notification_test.go]: Done.

**Comment by vish9812** [comment]
@jgilliam17 It's up for testing.

**Comment by jgilliam17** [comment]
Thanks @vish9812 
Looks great 
I have few observations after first test pass
1. I don't see the alert component when channel is muted. Is it supposed to be shown?
2. We display two options bearing `default`  wording in Desktop and Mobile notifications sections. This may be confusing for users

**Comment by vish9812** [comment]
@jgilliam17
* I don't know which alert component you're referring to and in which scenario it should show or not show?
* This PR is specifically for the "Auto-follow" feature. @AshishDhama can help you out with the Modla/UI related issues. Here is the [Web PR](https://github.com/mattermost/mattermost-webapp/pull/11515) for it.

**Review by koox00 — APPROVED** [review]
LGTM 💪, do we need to have the feature behind a feature flag though?

**Review by kostaspt — APPROVED** [review]

**Review by AshishDhama — APPROVED** [review]

**Comment by vish9812** [comment]
@furqanmlk It's up for review.

**Comment by vish9812** [comment]
Hey @stevemudie is it looking good to merge?

**Comment by stevemudie** [comment]
> Hey @stevemudie is it looking good to merge?

@vish9812 Sorry for the delay. I wasn't able to get the E2E tests to complete prior to today but they have now and I'm looking the results over next. After that I just need to test it out on the Spinwick and it should be good to go.

It shouldn't be long! 😄

**Review by stevemudie — APPROVED** [review]
@vish9812 This is all set!

**Pull request merged**